src/ia-book-actions.js

```diff
@@ -301,27 +307,24 @@ export default class IABookActions extends LitElement {
     this.loanRenewResult = this.loanRenewHelper.result;
   }
 
-  /**
-   * create / select the modal-manager component on DOM
-   * @memberof IABookActions
-   */
-  async useModalManager() {
-    if (!this.modal) {
-      this.modal = document.querySelector('modal-manager');
-
-      if (!this.modal) this.modal = document.createElement('modal-manager');
-    }
+  /** @returns HTMLElement */
+  get modal() {
+    const modalOnDom = document.querySelector('modal-manager');
 
-    this.modal.id = 'action-bar-modal';
-    await document.body.appendChild(this.modal);
+    modalOnDom.id = 'action-bar-modal';
+    return modalOnDom;
   }
 
   /**
    * Show the waring modal to ask user if they are still reading
    */
   async showWarningModal() {
     console.log('****** showWarningModal ******');
-    await this.useModalManager();
+    // clear modal
+    this.modal.customModalContent = nothing;
```

Comment:
@iisa above we are gettiing return modalOnDom but here still using this.modal object? are we still need this.modal object?

Reply:
yes i removed class property and made getter.